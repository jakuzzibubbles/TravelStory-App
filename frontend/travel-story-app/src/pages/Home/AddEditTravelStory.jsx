import React from "react";
import { useState } from "react";
import { MdAdd, MdUpdate, MdClose } from "react-icons/md";
import DateSelector from "../../components/Input/DateSelector";
import ImageSelector from "../../components/Input/ImageSelector";
import TagInput from "../../components/Input/TagInput";
import axiosInstance from "../../utils/axiosInstance";
import uploadImage from "../../utils/uploadImage";
import { toast } from "react-toastify";
import moment from "moment";

const AddEditTravelStory = ({
  storyInfo,
  type,
  onClose,
  getAllTravelStories,
}) => {
  const [form, setForm] = useState({
    title: storyInfo?.title || "",
    story: storyInfo?.story || "",
    tags: storyInfo?.tags || [],
    imageUrl: storyInfo?.imageUrl || "",
    visitedLocation: storyInfo?.visitedLocation || "",
    visitedDate: storyInfo?.visitedDate || null,
  });

  // Add New Travel Story
  const addNewTravelStory = async () => {
    try {
      let imageUrl = "";

      // Upload image if present
      if (storyImg) {
        const imgUploadRes = await uploadImage(storyImg);
        // Get image URL
        imageUrl = imgUploadRes.imageUrl || "";
      }

      const response = await axiosInstance.post("/api/add-travel-story", {
        title,
        story,
        imageUrl: imageUrl || "",
        visitedLocation,
        visitedDate: visitedDate
          ? moment(visitedDate).valueOf()
          : moment().valueOf(),
      });

      if (response.data && response.data.story) {
        toast.success("Story added successfully");
        // Refresh stories
        getAllTravelStories();
        // Close Modal or Form
        onClose();
      }
    } catch (error) {
      if (
        error.response &&
        error.response.data &&
        error.response.data.message
      ) {
        setError(error.response.data.message);
      } else {
        setError("An unexpected error occurred. Please try again.");
      }
    }
  };

  // Update Travel Story
  const updateTravelStory = async () => {
    const storyId = storyInfo._id;
    try {
      let imageUrl = "";

      let postData = {
        title,
        story,
        imageUrl: storyInfo.imageUrl || "",
        visitedLocation,
        visitedDate: visitedDate
          ? moment(visitedDate).valueOf()
          : moment().valueOf(),
      };

      if (typeof storyImg === "object") {
        // Upload new image
        const imgUploadRes = await uploadImage(storyImg);
        imageUrl = imgUploadRes.imageUrl || "";

        postData = {
          ...postData,
          imageUrl: imageUrl,
        };
      }

      const response = await axiosInstance.put(
        "/api/edit-story/" + storyId,
        postData
      );

      if (response.data && response.data.story) {
        toast.success("Story updated successfully");
        // Refresh stories
        getAllTravelStories();
        // Close Modal or Form
        onClose();
      }
    } catch (error) {
      if (
        error.response &&
        error.response.data &&
        error.response.data.message
      ) {
        setError(error.response.data.message);
      } else {
        setError("An unexpected error occurred. Please try again.");
      }
    }
  };

  const handleAddOrUpdateClick = () => {
    console.log("Input Data:", {
      title,
      storyImg,
      story,
      visitedLocation,
      visitedDate,
    });

    if (!title) {
      setError("Please enter the title");
      return;
    }

    if (!story) {
      setError("Please enter the story");
      return;
    }

    setError("");

    if (type === "edit") {
      updateTravelStory();
    } else {
      addNewTravelStory();
    }
  };

  // Delete story image and update story
  const handleDeleteStoryImg = async () => {
    const deleteImgRes = await axiosInstance.delete("/api/delete-image", {
      params: {
        imageUrl: storyInfo.imageUrl,
      },
    });

    if (deleteImgRes.data) {
      const storyId = storyInfo._id;

      const postData = {
        title,
        story,
        visitedLocation,
        visitedDate: moment().valueOf(),
        imageUrl: "",
      };

      // Updating story
      await axiosInstance.put("/api/edit-story/" + storyId, postData);
      setStoryImg(null);
    }
  };

  return (
    <div className="relative">
      <div className="flex items-center justify-between">
        <h5 className="text-xl font-medium text-slate-700">
          {type === "add" ? "Add Story" : "Update Story"}
        </h5>

        <div>
          <div className="flex items-center gap-3 bg-cyan-50/50 p-2 rounded-l-lg">
            {type === "add" ? (
              <button className="btn-small" onClick={handleAddOrUpdateClick}>
                <MdAdd className="text-lg" /> ADD STORY
              </button>
            ) : (
              <>
                <button className="btn-small" onClick={handleAddOrUpdateClick}>
                  <MdUpdate className="text-lg" /> UPDATE STORY
                </button>
              </>
            )}

            <button className="" onClick={onClose}>
              <MdClose className="text-xl text-slate-400" />
            </button>
          </div>

          {error && (
            <p className="text-red-500 text-xs pt-2 text-right">{error}</p>
          )}
        </div>
      </div>

      <div>
        <div className="flex-1 flex flex-col gap-2 pt-4">
          <label className="input-label">TITLE</label>
          <input
            type="text"
            className="text-2xl text-slate-950 outline-none"
            placeholder="A Day in Space"
            value={title}
            onChange={({ target }) => setTitle(target.value)}
          />

          <div className="my-3">
            <DateSelector date={visitedDate} setDate={setVisitedDate} />
          </div>

          <ImageSelector
            image={storyImg}
            setImage={setStoryImg}
            handleDeleteImg={handleDeleteStoryImg}
          />

          <div className="flex flex-col gap-2 mt-4">
            <label className="input-label">STORY</label>
            <textarea
              type="text"
              className="text-sm text-slate-950 outline-none bg-slate-50 p-2 rounded"
              placeholder="Your Story"
              rows={10}
              value={story}
              onChange={({ target }) => setStory(target.value)}
            />
          </div>

          <div className="pt-3">
            <label className="input-label">VISITED LOCATION</label>
            <TagInput tags={visitedLocation} setTags={setVisitedLocation} />
          </div>
        </div>
      </div>
    </div>
  );
};

export default AddEditTravelStory;