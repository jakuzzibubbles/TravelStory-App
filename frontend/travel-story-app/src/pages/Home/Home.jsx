import { useEffect, useState } from "react";
import Navbar from "../../components/Navbar";
import { useNavigate } from "react-router-dom";
import axiosInstance from "../../utils/axiosInstance";
import { MdAdd } from "react-icons/md";
import Modal from "react-modal";
import TravelStoryCard from "../../components/Cards/TravelStoryCard";
import AddEditTravelStory from "./AddEditTravelStory";
import ViewTravelStory from "./ViewTravelStory";
import EmptyCard from "../../components/Cards/EmptyCard";
import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import { DayPicker } from "react-day-picker";
import moment from "moment";
import FilterInfoTitle from "../../components/Cards/FilterInfoTitle";
import { getEmptyCardImg, getEmptyCardMessage } from "../../utils/helper";

const Home = () => {
  const navigate = useNavigate();

  const [userInfo, setUserInfo] = useState(null);
  const [allStories, setAllStories] = useState([]);
  const [searchQuery, setSearchQuery] = useState("");
  const [filterType, setFilterType] = useState("");
  const [dateRange, setDateRange] = useState({ from: null, to: null });

  const [openAddEditModal, setOpenAddEditModal] = useState({
    isShown: false,
    type: "add",
    data: null,
  });

  const [openViewModal, setOpenViewModal] = useState({
    isShown: false,
    data: null,
  });

  // Define resetFilter function
  const resetFilter = () => {
    setSearchQuery("");
    setFilterType("");
    setDateRange({ from: null, to: null });
    getAllTravelStories();
  };

  // Fetch User Info
  const getUserInfo = async () => {
    try {
      const { data } = await axiosInstance.get("/user");
      if (data?.user) setUserInfo(data.user);
    } catch (error) {
      if (error.response?.status === 401) {
        localStorage.clear();
        navigate("/login");
      }
    }
  };

  // Fetch All Travel Stories
  const getAllTravelStories = async () => {
    try {
      const { data } = await axiosInstance.get("/stories");
      if (data?.stories) setAllStories(data.stories);
    } catch (error) {
      console.error("Failed to fetch stories. Please try again.");
    }
  };

  // Toggle Favourite Status
  const updateIsFavourite = async (storyData) => {
    try {
      const { data } = await axiosInstance.put(
        `/stories/${storyData._id}/favourite`,
        {
          isFavourite: !storyData.isFavourite,
        }
      );

      if (data?.story) {
        toast.success("Story updated successfully.");
        refreshStories();
      }
    } catch (error) {
      console.error("Failed to update favourite status. Please try again.");
    }
  };

  // Delete Story
  const deleteTravelStory = async (data) => {
    try {
      const { data: response } = await axiosInstance.delete(
        `/stories/${data._id}`
      );
      if (response?.success) {
        toast.success("Story deleted successfully.");
        setOpenViewModal({ isShown: false });
        getAllTravelStories();
      }
    } catch (error) {
      console.error("Failed to delete story. Please try again.");
    }
  };

  // Search Story
  const onSearchNote = async (query) => {
    try {
      const { data } = await axiosInstance.get("/stories/search", {
        params: { query },
      });
      if (data?.stories) {
        setFilterType("search");
        setAllStories(data.stories);
      }
    } catch (error) {
      console.error("Failed to search stories. Please try again.");
    }
  };

  const handleClearSearch = () => {
    setFilterType("");
    getAllTravelStories();
  };

  // Filter Stories by Date
  const filterStoriesByDate = async (day) => {
    try {
      const startDate = day.from ? moment(day.from).valueOf() : null;
      const endDate = day.to ? moment(day.to).valueOf() : null;

      if (startDate && endDate) {
        const { data } = await axiosInstance.get("/stories/filter", {
          params: { startDate, endDate },
        });
        if (data?.stories) {
          setFilterType("date");
          setAllStories(data.stories);
        }
      }
    } catch (error) {
      console.error("Failed to filter stories by date. Please try again.");
    }
  };

  // Handle Day Click for Date Picker
  const handleDayClick = (day) => {
    setDateRange({ from: day.from, to: day.to });
    filterStoriesByDate({ from: day.from, to: day.to });
  };

  // Refresh Stories based on Filters
  const refreshStories = () => {
    if (filterType === "search" && searchQuery) {
      onSearchNote(searchQuery);
    } else if (filterType === "date") {
      filterStoriesByDate(dateRange);
    } else {
      getAllTravelStories();
    }
  };

  useEffect(() => {
    getAllTravelStories();
    getUserInfo();
  }, []);

  return (
    <>
      <Navbar
        userInfo={userInfo}
        searchQuery={searchQuery}
        setSearchQuery={setSearchQuery}
        onSearchNote={onSearchNote}
        handleClearSearch={handleClearSearch}
      />

      <div className="container mx-auto py-10">
        <FilterInfoTitle
          filterType={filterType}
          filterDates={dateRange}
          onClear={resetFilter} // Use the resetFilter function
        />

        <div className="flex gap-7">
          <div className="flex-1">
            {allStories?.length > 0 ? (
              <div className="grid grid-cols-2 gap-4">
                {allStories.map((item) => (
                  <TravelStoryCard
                    key={item._id}
                    imgUrl={item.imageUrl}
                    title={item.title}
                    story={item.story}
                    date={item.visitedDate}
                    visitedLocation={item.visitedLocation}
                    isFavourite={item.isFavourite}
                    onClick={() => handleViewStory(item)}
                    onFavouriteClick={() => updateIsFavourite(item)}
                  />
                ))}
              </div>
            ) : (
              <EmptyCard
                imgSrc={getEmptyCardImg(filterType)}
                message={getEmptyCardMessage(filterType)}
              />
            )}
          </div>

          <div className="w-[350px]">
            <div className="bg-white border border-slate-200 shadow-lg shadow-slate-200/60 rounded-lg">
              <div className="p-3">
                <DayPicker
                  captionLayout="dropdown-buttons"
                  mode="range"
                  selected={dateRange}
                  onSelect={handleDayClick} // Set the onSelect prop
                  pagedNavigation
                />
              </div>
            </div>
          </div>
        </div>
      </div>

      {/* Add & Edit Travel Story Modal */}
      <Modal
        isOpen={openAddEditModal.isShown}
        onRequestClose={() => {}}
        style={{
          overlay: {
            backgroundColor: "rgba(0,0,0,0.2)",
            zIndex: 999,
          },
        }}
        appElement={document.getElementById("root")}
        className="model-box"
      >
        <AddEditTravelStory
          type={openAddEditModal.type}
          storyInfo={openAddEditModal.data}
          onClose={() => {
            setOpenAddEditModal({ isShown: false, type: "add", data: null });
          }}
          getAllTravelStories={getAllTravelStories}
        />
      </Modal>

      {/* View Travel Story Modal */}
      <Modal
        isOpen={openViewModal.isShown}
        onRequestClose={() => {}}
        style={{
          overlay: {
            backgroundColor: "rgba(0,0,0,0.2)",
            zIndex: 999,
          },
        }}
        appElement={document.getElementById("root")}
        className="model-box"
      >
        <ViewTravelStory
          storyInfo={openViewModal.data || null}
          onClose={() => {
            setOpenViewModal({ isShown: false });
          }}
          onEditClick={() => {
            setOpenViewModal({ isShown: false });
            handleEdit(openViewModal.data || null);
          }}
          onDeleteClick={() => {
            deleteTravelStory(openViewModal.data || null);
          }}
        />
      </Modal>

      <button
        className="w-16 h-16 flex items-center justify-center rounded-full bg-primary hover:bg-cyan-400 fixed right-10 bottom-10"
        onClick={() => {
          setOpenAddEditModal({ isShown: true, type: "add", data: null });
        }}
      >
        <MdAdd className="text-[32px] text-white" />
      </button>

      <ToastContainer />

      {/* Display User Initials */}
      {userInfo && (
        <div className="absolute top-4 right-4 text-xl">
          {userInfo.firstName.charAt(0)}
          {userInfo.lastName.charAt(0)}
        </div>
      )}
    </>
  );
};

export default Home;
