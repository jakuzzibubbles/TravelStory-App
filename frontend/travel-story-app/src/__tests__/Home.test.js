import React from "react";
import Home from "../pages/Home/Home.jsx";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// Mock dependencies
jest.mock("react-toastify", () => ({
  ToastContainer: () => null,
}));

jest.mock("../components/Navbar.jsx", () => () => <div>Navbar</div>);
jest.mock("../components/Cards/TravelStoryCard.jsx", () => () => (
  <div>TravelStoryCard</div>
));
jest.mock("../components/Cards/EmptyCard.jsx", () => () => <div>EmptyCard</div>);
jest.mock("../components/Cards/FilterInfoTitle.jsx", () => () => (
  <div>FilterInfoTitle</div>
));
jest.mock("../pages/Home/AddEditTravelStory.jsx", () => () => <div>AddEditTravelStory</div>);
jest.mock("../pages/Home/ViewTravelStory.jsx", () => () => <div>ViewTravelStory</div>);

describe("Home Component Snapshot", () => {
  it("renders correctly", () => {
    const { container } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
