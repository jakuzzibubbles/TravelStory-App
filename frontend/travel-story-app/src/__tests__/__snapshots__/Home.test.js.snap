// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Snapshot renders correctly 1`] = `
<div>
  <div>
    Navbar
  </div>
  <div
    class="container mx-auto py-10"
  >
    <div>
      FilterInfoTitle
    </div>
    <div
      class="flex gap-7"
    >
      <div
        class="flex-1"
      >
        <div>
          EmptyCard
        </div>
      </div>
      <div
        class="w-[350px]"
      >
        <div
          class="bg-white border border-slate-200 shadow-lg shadow-slate-200/60 rounded-lg"
        >
          <div
            class="p-3"
          >
            <div
              class="rdp-root"
              data-mode="range"
            >
              <div
                class="rdp-months"
              >
                <nav
                  aria-label=""
                  class="rdp-nav"
                >
                  <button
                    aria-label="Go to the Previous Month"
                    class="rdp-button_previous"
                    type="button"
                  >
                    <svg
                      class="rdp-chevron"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                    >
                      <polygon
                        points="16 18.112 9.81111111 12 16 5.87733333 14.0888889 4 6 12 14.0888889 20"
                      />
                    </svg>
                  </button>
                  <button
                    aria-label="Go to the Next Month"
                    class="rdp-button_next"
                    type="button"
                  >
                    <svg
                      class="rdp-chevron"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                    >
                      <polygon
                        points="8 18.612 14.1888889 12.5 8 6.37733333 9.91111111 4.5 18 12.5 9.91111111 20.5"
                      />
                    </svg>
                  </button>
                </nav>
                <div
                  class="rdp-month"
                >
                  <div
                    class="rdp-month_caption"
                  >
                    <div
                      class="rdp-dropdowns"
                    >
                      <span
                        aria-live="polite"
                        role="status"
                      >
                        November
                      </span>
                      <span
                        aria-live="polite"
                        role="status"
                      >
                        2024
                      </span>
                    </div>
                  </div>
                  <table
                    aria-label="November 2024"
                    aria-multiselectable="true"
                    class="rdp-month_grid"
                    role="grid"
                  >
                    <thead>
                      <tr
                        class="rdp-weekdays"
                      >
                        <th
                          aria-label="Sunday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          Su
                        </th>
                        <th
                          aria-label="Monday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          Mo
                        </th>
                        <th
                          aria-label="Tuesday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          Tu
                        </th>
                        <th
                          aria-label="Wednesday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          We
                        </th>
                        <th
                          aria-label="Thursday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          Th
                        </th>
                        <th
                          aria-label="Friday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          Fr
                        </th>
                        <th
                          aria-label="Saturday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          Sa
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="rdp-weeks"
                    >
                      <tr
                        class="rdp-week"
                      >
                        <td
                          aria-hidden="true"
                          class="rdp-day rdp-hidden rdp-outside"
                          data-day="2024-10-27"
                          data-hidden="true"
                          data-month="2024-10"
                          data-outside="true"
                        >
                          <button
                            aria-label="Sunday, October 27th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            27
                          </button>
                        </td>
                        <td
                          aria-hidden="true"
                          class="rdp-day rdp-hidden rdp-outside"
                          data-day="2024-10-28"
                          data-hidden="true"
                          data-month="2024-10"
                          data-outside="true"
                        >
                          <button
                            aria-label="Monday, October 28th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            28
                          </button>
                        </td>
                        <td
                          aria-hidden="true"
                          class="rdp-day rdp-hidden rdp-outside"
                          data-day="2024-10-29"
                          data-hidden="true"
                          data-month="2024-10"
                          data-outside="true"
                        >
                          <button
                            aria-label="Tuesday, October 29th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            29
                          </button>
                        </td>
                        <td
                          aria-hidden="true"
                          class="rdp-day rdp-hidden rdp-outside"
                          data-day="2024-10-30"
                          data-hidden="true"
                          data-month="2024-10"
                          data-outside="true"
                        >
                          <button
                            aria-label="Wednesday, October 30th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            30
                          </button>
                        </td>
                        <td
                          aria-hidden="true"
                          class="rdp-day rdp-hidden rdp-outside"
                          data-day="2024-10-31"
                          data-hidden="true"
                          data-month="2024-10"
                          data-outside="true"
                        >
                          <button
                            aria-label="Thursday, October 31st, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            31
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-01"
                        >
                          <button
                            aria-label="Friday, November 1st, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            1
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-02"
                        >
                          <button
                            aria-label="Saturday, November 2nd, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            2
                          </button>
                        </td>
                      </tr>
                      <tr
                        class="rdp-week"
                      >
                        <td
                          class="rdp-day"
                          data-day="2024-11-03"
                        >
                          <button
                            aria-label="Sunday, November 3rd, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            3
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-04"
                        >
                          <button
                            aria-label="Monday, November 4th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            4
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-05"
                        >
                          <button
                            aria-label="Tuesday, November 5th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            5
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-06"
                        >
                          <button
                            aria-label="Wednesday, November 6th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            6
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-07"
                        >
                          <button
                            aria-label="Thursday, November 7th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            7
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-08"
                        >
                          <button
                            aria-label="Friday, November 8th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            8
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-09"
                        >
                          <button
                            aria-label="Saturday, November 9th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            9
                          </button>
                        </td>
                      </tr>
                      <tr
                        class="rdp-week"
                      >
                        <td
                          class="rdp-day"
                          data-day="2024-11-10"
                        >
                          <button
                            aria-label="Sunday, November 10th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            10
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-11"
                        >
                          <button
                            aria-label="Monday, November 11th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            11
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-12"
                        >
                          <button
                            aria-label="Tuesday, November 12th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            12
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-13"
                        >
                          <button
                            aria-label="Wednesday, November 13th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            13
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-14"
                        >
                          <button
                            aria-label="Thursday, November 14th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            14
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-15"
                        >
                          <button
                            aria-label="Friday, November 15th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            15
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-16"
                        >
                          <button
                            aria-label="Saturday, November 16th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            16
                          </button>
                        </td>
                      </tr>
                      <tr
                        class="rdp-week"
                      >
                        <td
                          class="rdp-day"
                          data-day="2024-11-17"
                        >
                          <button
                            aria-label="Sunday, November 17th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            17
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-18"
                        >
                          <button
                            aria-label="Monday, November 18th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            18
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-19"
                        >
                          <button
                            aria-label="Tuesday, November 19th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            19
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-20"
                        >
                          <button
                            aria-label="Wednesday, November 20th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            20
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-21"
                        >
                          <button
                            aria-label="Thursday, November 21st, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            21
                          </button>
                        </td>
                        <td
                          class="rdp-day rdp-today"
                          data-day="2024-11-22"
                          data-today="true"
                        >
                          <button
                            aria-label="Today, Friday, November 22nd, 2024"
                            class="rdp-day_button"
                            tabindex="0"
                            type="button"
                          >
                            22
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-23"
                        >
                          <button
                            aria-label="Saturday, November 23rd, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            23
                          </button>
                        </td>
                      </tr>
                      <tr
                        class="rdp-week"
                      >
                        <td
                          class="rdp-day"
                          data-day="2024-11-24"
                        >
                          <button
                            aria-label="Sunday, November 24th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            24
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-25"
                        >
                          <button
                            aria-label="Monday, November 25th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            25
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-26"
                        >
                          <button
                            aria-label="Tuesday, November 26th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            26
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-27"
                        >
                          <button
                            aria-label="Wednesday, November 27th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            27
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-28"
                        >
                          <button
                            aria-label="Thursday, November 28th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            28
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-29"
                        >
                          <button
                            aria-label="Friday, November 29th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            29
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2024-11-30"
                        >
                          <button
                            aria-label="Saturday, November 30th, 2024"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            30
                          </button>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    class="w-16 h-16 flex items-center justify-center rounded-full bg-primary hover:bg-cyan-400 fixed right-10 bottom-10"
  >
    <svg
      class="text-[32px] text-white"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </button>
</div>
`;
